core/src/proof.rs

```diff
@@ -16,11 +17,40 @@ pub struct DataIntegrityProof {
     pub proof_value: String,
 }
 
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone, PartialEq, Eq)]
+pub struct RsaSignature2018 {
+    #[serde(rename = "type")]
+    pub proof_type: String,
+    #[serde(rename = "created")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub created: Option<String>,
+    #[serde(rename = "verificationMethod")]
+    pub verification_method: String,
+    #[serde(rename = "proofPurpose")]
+    pub proof_purpose: String,
+    #[serde(rename = "jws")]
+    pub jws: String,
+}
+
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum CredentialProof {
```

Comment:
Why not just have this be a type alias for `Vec<ProofType>` and allow the vec to contain one thing? Do we need a way to statically guarantee that there are never 0 credential proofs?

Reply:
@gshuflin Credentials proofs can be structured as either a array of objects or just a singular object. And we need to be able to parse them and handle them in either scenario.

If we only expect an array we won't be able to parse the proofs where its just an object. This is part of the nature of the VC spec